ecs-cli/modules/aws/clients/ecr/client.go

```diff
@@ -117,3 +123,90 @@ func (c *ecrClient) CreateRepository(repositoryName string) (string, error) {
 	log.Info("Repository created")
 	return aws.StringValue(resp.Repository.RepositoryName), nil
 }
+
+func (c *ecrClient) GetImages(repositoryNames []*string, tagStatus string, registryID string, processFn ProcessImageDetails) error {
+	log.Debug("Getting images")
+	numOfImageCalls := 0
+	err := c.describeRepositories(repositoryNames, registryID, func(repositories []*string) error {
+		for _, repository := range repositories {
+			err := c.describeImages(aws.StringValue(repository), tagStatus, registryID, processFn, numOfImageCalls)
+			numOfImageCalls++
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+
+	return err
+}
+
+func (c *ecrClient) describeRepositories(repositoryNames []*string, registryID string, outputFn ProcessRepositories) error {
+	var outErr error
+
+	input := &ecr.DescribeRepositoriesInput{}
+
+	// Skip DescribeRepositories calls if repositoryNames are specified
+	if repositoryNames != nil && len(repositoryNames) > 0 {
+		if err := outputFn(repositoryNames); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	if registryID != "" {
+		input.SetRegistryId(registryID)
+	}
+
+	err := c.client.DescribeRepositoriesPages(input, func(resp *ecr.DescribeRepositoriesOutput, lastPage bool) bool {
+		repositoryNames = []*string{}
+		for _, repository := range resp.Repositories {
+			repositoryNames = append(repositoryNames, repository.RepositoryName)
+		}
+		if err := outputFn(repositoryNames); err != nil {
+			outErr = err
+			return false
+		}
+		return !lastPage
+	})
+
+	if err != nil {
+		return err
+	}
+	return outErr
```

Comment:
This is to throw an error we got an error from the API call or we got an error from the callback function. Otherwise, how does the outer function know what the error is?